src/test/java/lambda/part3/exercise/Mapping.java

```diff
@@ -120,34 +146,51 @@ public LazyMapHelper(List<T> list, Function<T, R> function) {
 
         public List<R> force() {
             // TODO
-            throw new UnsupportedOperationException();
+            final List<R> result = new ArrayList<>();
+            list.forEach(t -> result.add(function.apply(t)));
+            return result;
         }
 
         public <R2> LazyMapHelper<T, R2> map(Function<R, R2> f) {
             // TODO
-            throw new UnsupportedOperationException();
+            return new LazyMapHelper<T, R2>(list, function.andThen(f));
         }
 
     }
 
     private static class LazyFlatMapHelper<T, R> {
+        final List<T> list;
+        final Function<T, List<R>> function;
 
         public LazyFlatMapHelper(List<T> list, Function<T, List<R>> function) {
+            this.list = list;
+            this.function = function;
         }
 
         public static <T> LazyFlatMapHelper<T, T> from(List<T> list) {
-            throw new UnsupportedOperationException();
+            return new LazyFlatMapHelper<T, T>(list, Arrays::asList);
         }
 
         public List<R> force() {
             // TODO
-            throw new UnsupportedOperationException();
+            final List<R> result = new ArrayList<R>();
+            list.forEach(t -> function.apply(t).forEach(result::add));
+            return result;
         }
 
         // TODO filter
         // (T -> boolean) -> (T -> [T])
         // filter: [T1, T2] -> (T -> boolean) -> [T2]
         // flatMap": [T1, T2] -> (T -> [T]) -> [T2]
+        public LazyFlatMapHelper<T, R> filter(Predicate<T> predicate) {
+            final List<T> result = new ArrayList<T>();

```

Comment:
Call `flatMap` here. You should not create a new `ArrayList` in `filter` and `map` methods.